### PR TITLE
Prepare for 2.0.8.1-RELEASE

### PR DESCRIPTION
**Pull request by JayantJain96**
- serialize and deserialize extensions with ref in schema

### CODE REVIEW

File: modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/SchemaSerializer.java [hunk 1]
```diff
@@ -37,6 +37,11 @@ public void serialize(
         } else {
             jgen.writeStartObject();
             jgen.writeStringField("$ref", value.get$ref());
+            if (value.getExtensions() != null) {
+              for (Object ext: value.getExtensions().keySet()) {
+                jgen.writeObjectField((String) ext , value.getExtensions().get(ext));
+              }
+            }
```
Comment (JayantJain96) [inline modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/SchemaSerializer.java:44]: serialize extensions when used with $ref in schemas

File: modules/swagger-core/src/main/java/io/swagger/v3/core/util/ModelDeserializer.java [hunk 2]
```diff
@@ -77,6 +78,10 @@ public Schema deserialize(JsonParser jp, DeserializationContext ctxt)
                 }
             } else if (node.get("$ref") != null) {
                 schema = new Schema().$ref(node.get("$ref").asText());
+                Map<String,Object> extensions = getExtensions(node);
+                if(extensions != null && extensions.size() > 0) {
+                  schema.setExtensions(extensions);
+                }
```
Comment (JayantJain96) [inline modules/swagger-core/src/main/java/io/swagger/v3/core/util/ModelDeserializer.java:84]: deserialize extensions when used with $ref in schemas.
We dont use this functionality in the platform, but its good to have the extensions feature supported in both ser/deser

Comment (JayantJain96) [reply modules/swagger-core/src/main/java/io/swagger/v3/core/util/ModelDeserializer.java:84]: Implementing for just one usecase (serialization) seemed incomplete

File: modules/swagger-core/src/test/java/io/swagger/v3/core/serialization/ModelSerializerTest.java [hunk 3]
```diff
@@ -106,6 +106,21 @@ public void deserializeModel() throws IOException {
...
+        assertEquals(m.writeValueAsString(model), json);
+    }
+
+    @Test(description = "it should deserialize a model with ref and extensions")
+    public void deserializeModelWithRefAndExtensions() throws IOException {
+        String json = "{\"$ref\":\"#/components/schemas/foo\",\"x-ext\":\"bar\"}";
+        Schema model = m.readValue(json, Schema.class);
+        assertEquals(m.writeValueAsString(model), json);
+    }
+
```
Comment (JayantJain96) [inline modules/swagger-core/src/test/java/io/swagger/v3/core/serialization/ModelSerializerTest.java:123]: corresponding tests

**Review by krraghavan — APPROVED** [review]

**Pull request merged**